File: python/贪心/1196. 最多可以买到的苹果数量.py
from typing import List
class Solution:
    def maxNumberOfApples(self, arr: List[int]) -> int:
        arr.sort()
        res = 5000
        n = 0
        for i in arr:
            if i > res:
                return n
            else:
                res -= i
                n+=1
        return n

File: go/greedy/1196. 最多可以买到的苹果数量.go
package greedy

import "sort"

func maxNumberOfApples(arr []int) int {
	sort.Ints(arr)
	n := 0
	ans := 5000
	for _, v := range arr {
		if v > ans {
			return n
		} else {
			ans -= v
			n++
		}
	}
	return n
}
